sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java

```diff
@@ -56,14 +52,12 @@
 import org.joda.time.Instant;
 import org.joda.time.MutableDateTime;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 /**
- * Tests for <a href="https://s.apache.org/splittable-do-fn>splittable</a> {@link DoFn} behavior
- * using the direct runner.
- *
- * <p>TODO: make this use @RunnableOnService.
+ * Tests for <a href="https://s.apache.org/splittable-do-fn>splittable</a> {@link DoFn} behavior.
```

Comment:
As a possible follow-up, tests really are more discoverable when they are found in `FooTest` when testing `Foo`. Sometimes it is a gray area, but these are clearly part of `ParDoTest`. I generally expect that running `ParDoTest` is going to get me pretty high confidence that my runner implements `ParDo` fully and correctly.

I believe you can use JUnit suites to write tests in a separate class like this for readability, but then incorporate them into the actual test suite.